eng/pipelines/templates/Build_and_UnitTest.yml

```diff
@@ -232,6 +232,15 @@ steps:
     msbuildArguments: "/t:EnsurePackagesExist /p:ExcludeTestProjects=$(BuildRTM)"
   condition: and(succeeded(), or(eq(variables['IsOfficialBuild'], 'true'), eq(variables['BuildRTM'], 'true')))  #skip this task for nonRTM in private build
 
+  # The list of nupkgs in the script might change, so we publish it as artifact, in order to make sure release pipeline check the right list.
+- task: PublishBuildArtifacts@1
+  displayName: "Publish EnsureAllPackagesExist script"
+  inputs:
+    PathtoPublish: "$(Build.Repository.LocalPath)\\scripts\\utils\\EnsureAllPackagesExist.ps1"
+    ArtifactName: "Scripts"
+    ArtifactType: "Container"
+  condition: "and(succeeded(),eq(variables['BuildRTM'], 'false'))"
```

Comment:
```suggestion
  condition: "and(succeeded(), eq(variables['BuildRTM'], 'false'))"
```

Reply:
Thanks for reviewing!
I investigated the other way of getting script from a specific commit hash and it should be workable.
We can either get commit hash from the variable you mentioned above, or from the buildinfo.json
Then according to [getting-permanent-links-to-files](https://docs.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files), we can reference it like: https://github.com/NuGet/NuGet.Client/blob/18863da5be3dc8c7315f4416df1bc9ef96cb7446/scripts/utils/EnsureAllPackagesExist.ps1 

Comparing the two ways, there is no difference for future builds. 
For servicing old branches without the script EnsureAllPackagesExist.ps1, it looks like it's a difference of cherry-picking one commit or two commits. Any thoughts?

Reply:
I lean towards using `commit hash` to read `EnsureAllPackagesExist.ps1` file instead of publishing the script as a build artifact.